yarn-project/acir-simulator/src/acvm/oracle/oracle.ts

```diff
@@ -163,9 +164,15 @@ export class Oracle {
     return toACVMField(0);
   }
 
-  emitUnencryptedLog(message: ACVMField[]): ACVMField {
+  emitUnencryptedLog([contractAddress]: ACVMField[], [eventSelector]: ACVMField[], message: ACVMField[]): ACVMField {
     // https://github.com/AztecProtocol/aztec-packages/issues/885
-    const log = Buffer.concat(message.map(charBuffer => convertACVMFieldToBuffer(charBuffer).subarray(-1)));
+    const logPayload = Buffer.concat(message.map(charBuffer => convertACVMFieldToBuffer(charBuffer).subarray(-1)));
+    const log = new UnencryptedL2Log(
+      AztecAddress.fromString(contractAddress),
+      FunctionSelector.fromField(fromACVMField(eventSelector)),
```

Comment:
What's the reason for using a FunctionSelector here? I think a plain `Buffer` (of a hash?) would be a better fit, but I might be wrong.

Reply:
We decided to not use raw buffers much because it turned out to be a pain to ensure that the data is consistent --> you would need to do length checks everywhere so it's better to have a special class for that which does the checks.

If you would not do the checks then there is a danger that you would somewhere for whatever reason end up with zero value selector and then you would have undefined behavior and the resulting bug might be unpredictable. i spent 4 hours debugging our SDK in Aztec Connect because nodejs does not throw an error when you try to initialize a buffer with a hex string with 0x at the beginning and I though my SDK modifications are messed up: 
<img width="741" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/13470840/0d4901b7-34a6-4d04-a330-fd107759a386">

Reply:
Yep, makes sense. I was confused by the naming :)